tests/rbd/rbd_snapshot_rename_advanced.py

```diff
@@ -0,0 +1,214 @@
+
+        results = {}
+
+        log.info("Rename the snapshot when IO on Clone is in progress. (use rbd bench)")
+        with parallel() as p:
+            image_spec = f"{pool}/{clone}"
+            p.spawn(
+                execute,
+                rbd,
+                "benchwrite",
+                results,
+                imagespec=image_spec,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=snap,
+                new_snap_name=snap_rename,
+            )
+
+        if results["snap_rename"] or results["benchwrite"]:
+            log.error(f"Renaming snapshot while writing IOs failed for {snap_name}")
+            return 1
+
+        log.info(
+            "Rename the snapshot, when resize operation is happening on Clone image."
+        )
+        with parallel() as p:
+            p.spawn(
+                execute,
+                rbd,
+                "image_resize",
+                results,
+                pool_name=pool,
+                image_name=clone,
+                size=resize,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=snap_rename,
+                new_snap_name=f"{snap_rename}_1",
+            )
+
+        if results["snap_rename"] or results["image_resize"]:
+            log.error(
+                f"Renaming snapshot while failed for {snap_name} while performing resize on clone {clone}"
+            )
+            return 1
+
+        log.info("Rename a snapshot, and then create a clone on top of it.")
+        clone_2 = f"{clone}_2"
+        snap_name = f"{pool}/{image}@{snap_rename}_1"
+        if rbd.create_clone(snap_name, pool, clone_2):
+            log.error(f"Creation of clone {clone_2} failed")
+            return 1
+
+        log.info(
+            "Rename the Parent snapshot, when Flatten operation is happening on Clone image."
+        )
+        with parallel() as p:
+            p.spawn(
+                execute,
+                rbd,
+                "flatten_clone",
+                results,
+                pool_name=pool,
+                image_name=clone_2,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=f"{snap_rename}_1",
+                new_snap_name=f"{snap_rename}_2",
+            )
+
+        if results["snap_rename"] or results["flatten_clone"]:
+            log.error(
+                f"Renaming snapshot while failed for {snap_name} while performing flatten on clone {clone_2}"
```

Comment:
'while' is misplaced in sentence, please correct

Reply:
Done.